### PR TITLE
feat: add simple Ethereum token operations

### LINKED ISSUE #162: Add simple Ethereum operations to mint/deposit/stake

### PR DESCRIPTION
**Pull request by karlem**
Close #162

### CODE REVIEW

File: cmd/ethereum.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+		Use:   "ethereum",
+		Short: "Ethereum allows to call certain smart contracts on Ethereum network.",
+	}
+)
+
+func init() {
+	rootCmd.AddCommand(ethereumCmd)
+
+	ethereumCmd.PersistentFlags().StringVar(&ethereumOpts.address, "address", "", "address of the Ethereum network")
+	ethereumCmd.PersistentFlags().Int64Var(&ethereumOpts.chainID, "chain-id", 0, "address of the Ethereum network")
```
Comment (peterbarrow) [inline cmd/ethereum.go:29]: Should the comment be about the chain-id?

File: cmd/ethereum_deposit.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+		&defaultEthSyncTimeout,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to create token session for %s: %w", ethereumDepositOpts.assetAddress, err)
+	}
+
+	amount := big.NewInt(ethereumDepositOpts.amount)
+
+	if _, err := tokenSession.ApproveSync(bridgeAddr, amount); err != nil {
+		return fmt.Errorf("failef to approve asset amount to bridge: %w", err)
```
Comment (peterbarrow) [inline cmd/ethereum_deposit.go]:
```suggestion
		return fmt.Errorf("failed to approve asset amount to bridge: %w", err)
```

File: cmd/ethereum_mint.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	ethereumMintOpts struct {
+		ownerPrivateKey string
+		tokenAddress    string
+		toAddress       string
+		amount          int64
+	}
+
+	ethereumMintCmd = &cobra.Command{
+		Use:   "mint",
+		Short: "Mint allows to mint a token managed by a Base Faucet Token contract.",
```
Comment (peterbarrow) [inline cmd/ethereum_mint.go]:
```suggestion
		Short: "Mint allows a token to be minted by a Base Faucet Token contract.",
```

File: cmd/ethereum_stake.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		ownerPrivateKey string
+		bridgeAddress   string
+		assetAddress    string
+		vegaPubKey      string
+		amount          int64
+	}
+
+	ethereumStakeCmd = &cobra.Command{
+		Use:   "stake",
+		Short: "Stake allows to stake an asset to given Vega public key.",
```
Comment (peterbarrow) [inline cmd/ethereum_stake.go]:
```suggestion
		Short: "Stake allows an asset to be staked to a given Vega public key.",
```

**Review by peterbarrow — DISMISSED** [review]
Good to go once comments are addressed

**Comment by gordsport** [comment]
closing as already implemented

**Pull request closed without merging**